src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1057,7 +1060,49 @@ public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbExcep
     }
 
     return getDatasetStatus(relationKey);
+  }
 
+  /**
+   * @param relationKey the relationKey of the dataset to persist
+   * @return the queryID
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public long persistDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+    long queryID;
+
+    /* Mark the relation as is_persistent */
+    try {
+      catalog.markRelationPersistent(relationKey);
+    } catch (CatalogException e) {
+      throw new DbException(e);
+    }
+
+    /* Create the query plan for persist */
+    try {
+      Map<Integer, SubQueryPlan> workerPlans = new HashMap<>();
+      for (Integer workerId : getWorkersForRelation(relationKey, null)) {
+        String partitionName =
+            String.format("hdfs://vega.cs.washington.edu:8020/partition%s/%s/%s/%s", workerId, relationKey
```

Comment:
For readability, I'd like at least a hyphen between "partition" and the worker ID (or `workers/<workerId>`). Let's discuss in the student meeting but not bikeshed over it.